test/functional-test/functional-test-lib/taskTestRunner.ts

```diff
@@ -0,0 +1,80 @@
+import * as cp from 'child_process';
+import path = require('path');
+import process = require('process');
+
+export interface TaskInfo {
+  name: string;
+  path: string;
+  inputVariables?: EnvironmentVariableDefinition[];
+}
+
+export interface TaskResult {
+  processResult:cp.SpawnSyncReturns<string>;
+  outputEnvironmentVariable?: EnvironmentVariableDefinition;
+}
+
+export interface EnvironmentVariableDefinition {
+  name: string;
+  value: string;
+}
+
+export class TaskRunner {
+  private taskResult:cp.SpawnSyncReturns<string> | undefined = undefined;
+  taskInfo: TaskInfo;
+  taskDirectory: string;
+
+  constructor(taskInfo: TaskInfo, taskDirectory: string) {
+    this.taskInfo = taskInfo;
+    this.taskDirectory = taskDirectory;
+  }
+
+  runTask(): TaskResult{
+    const normalizedTaskPath = this.normalizeAbsoluteTaskPath()
+    this.taskResult = cp.spawnSync('node', [normalizedTaskPath], { encoding: 'utf-8', cwd: this.taskDirectory });
+    console.debug(this.taskResult.stdout);
+    this.validateTaskRun();
+    var envVar = this.setOutputEnvironmentVariables();
+    return {processResult: this.taskResult, outputEnvironmentVariable: envVar };
```

Comment:
Style nit: space after the opening brace
`return { processResult`
and perhaps some linting rules to enforce this, here or in another PR?

Reply:
Agreed, let's add some auto linting if possible, to ensure a general pattern / style is followed.